submissions/models.py

```diff
@@ -33,6 +33,11 @@
     providing_args=['anonymous_user_id', 'course_id', 'item_id']
 )
 
+# In order to allow for effective soft-deletes, we want to override the default Manager for several object types
+class SoftDeletedManager(models.Manager):
+    def get_queryset(self):
+        #By default, filter out any soft-deleted records for classes using this manager
+        return super(SoftDeletedManager, self).get_queryset().filter(deleted=False)
```

Comment:
@mulby, I'm curious if this approach to soft-deleting will meet everyone's needs. By overriding the `objects` manager for the relevant classes, we'll essentially be hiding the deleted records from any query internal to the djangoapp.

When the data is gathered for analytics, is it accessed using django models (e.g. `Submissions.objects.all()`), or taken straight from the database using other methods? Is the same true for instuctor-initiated data downloads?